### PR TITLE
Modified DTCWTForward

### LINKED ISSUE #2: Modification of DTCWTForward
- Modified the DTCWTForward class to make its results consistent with the [dtcwt(numpy) ](https://github.com/rjw57/dtcwt/tree/master/dtcwt/numpy)package.
- Added 2 parameters  to DTCWTForward: padding_mode and normalize
- Added a test function

### PR DESCRIPTION
**Pull request by xir4n**
resolves #2

### CODE REVIEW

File: murenn/dtcwt/utils.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    *maxx*.  Hence a vector *x* containing a long linearly increasing series is
+    converted into a waveform which ramps linearly up and down between *minx*
+    and *maxx*.  If *x* contains integers and *minx* and *maxx* are (integers +
+    0.5), the ramps will have repeated max and min samples.
+
+    .. codeauthor:: Rich Wareham <rjw57@cantab.net>, Aug 2013
+    .. codeauthor:: Nick Kingsbury, Cambridge University, January 1999.
+
+    """
+    x = np.asanyarray(x)
```
Comment (lostanlen) [inline murenn/dtcwt/utils.py:16]: i don't see a reason to use numpy here. try converting to torch?

File: murenn/dtcwt/utils.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def reflect(x, minx, maxx):
+    """Reflect the values in matrix *x* about the scalar values *minx* and
+    *maxx*.  Hence a vector *x* containing a long linearly increasing series is
+    converted into a waveform which ramps linearly up and down between *minx*
+    and *maxx*.  If *x* contains integers and *minx* and *maxx* are (integers +
+    0.5), the ramps will have repeated max and min samples.
+
+    .. codeauthor:: Rich Wareham <rjw57@cantab.net>, Aug 2013
```
Comment (lostanlen) [inline murenn/dtcwt/utils.py:12]: add yourself as codeauthor and say "adapted from:" then cite Wareham and Kingsbury

File: murenn/dtcwt/utils.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    return np.array(out, dtype=x.dtype)
+
+def symm_pad(l, m):
+    """ Creates indices for symmetric padding. Works for 1-D.
+
+    Inptus:
+        l (int): size of input
+        m (int): size of filter
+    """
+    xe = reflect(np.arange(-m, l+m, dtype='int32'), -0.5, l-0.5)
```
Comment (lostanlen) [inline murenn/dtcwt/utils.py:31]: again i feel like this could be pytorch

File: tests/test_dtcwt.py [hunk 4]
```diff
@@ -9,3 +10,18 @@
...
+    Xt = torch.tensor(X, dtype=torch.get_default_dtype()).view(1,1,2**(J+2))
+    xfm_murenn = murenn.DTCWTForward(J=J, alternate_gh=False, include_scale=False,padding_mode='reflect', normalize=False)
+    phis, psis = xfm_murenn(Xt)
+    xfm_np = dtcwt.Transform1d()
+    out_np = xfm_np.forward(X, nlevels=J)
+    phis_np = out_np.lowpass
+    psis_np = out_np.highpasses
+    np.testing.assert_array_almost_equal(phis_np[:,0], phis.numpy()[0,0,:], decimal = decimal)
+    for j in range(J):
+        np.testing.assert_array_almost_equal((psis_np[j])[:,0], psis[j].numpy()[0,0,:], decimal = decimal)
```
Comment (lostanlen) [inline tests/test_dtcwt.py:27]: amazing ...

File: murenn/dtcwt/utils.py (hunk 1 above)
Comment (xir4n) [reply murenn/dtcwt/utils.py]: done

File: murenn/dtcwt/utils.py (hunk 2 above)
Comment (xir4n) [reply murenn/dtcwt/utils.py]: done

File: tests/test_dtcwt.py (hunk 4 above)
Comment (lostanlen) [reply tests/test_dtcwt.py:27]: note: dtcwt only works with NumPy 3.10 or earlier.

File: murenn/dtcwt/utils.py (hunk 2 above)
Comment (lostanlen) [reply murenn/dtcwt/utils.py]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-codeauthor
for your information codeauthor:: is a Sphinx directive

**Review by lostanlen — APPROVED** [review]
